spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -75,46 +76,35 @@ protected SQLAdmin sqlAdminService(CredentialsProvider credentialsProvider)
 
 		return new SQLAdmin.Builder(httpTransport, jsonFactory,
 				new HttpCredentialsAdapter(credentialsProvider.getCredentials()))
-				.build();
+						.build();
 	}
 
 	@Bean
-	@ConditionalOnMissingBean(CloudSqlJdbcUrlProvider.class)
-	public CloudSqlJdbcUrlProvider cloudSqlJdbcUrlProvider(SQLAdmin sqlAdmin,
-			GcpProjectIdProvider projectIdProvider)
-			throws IOException, ClassNotFoundException {
-		Assert.hasText(projectIdProvider.getProjectId(),
-				"A project ID must be provided.");
-
-		// TODO(joaomartins): Set the credential factory to be used by SocketFactory when
-		// https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues/41 is
-		// resolved. For now, it uses application default creds.
-
-		// Load MySQL driver to register it, so the user doesn't have to.
-		Class.forName("com.mysql.jdbc.Driver");
-
-		// Look for the region if the user didn't specify one.
-		String region = this.gcpCloudSqlProperties.getRegion();
-		if (!StringUtils.hasText(region)) {
-			region = sqlAdmin.instances().get(projectIdProvider.getProjectId(),
-					this.gcpCloudSqlProperties.getInstanceName()).execute().getRegion();
-		}
-
-		String instanceConnectionName = projectIdProvider.getProjectId() + ":" + region + ":"
-				+ this.gcpCloudSqlProperties.getInstanceName();
+	@ConditionalOnMissingBean(CloudSqlJdbcInfoProvider.class)
+	@Conditional(AppEngineCondition.class)
```

Comment:
@artembilan added AppEngineCondition here, but just wnat to make sure this override makes sense, since we also have another ConditionalOnMissingBean just below it.